Microsoft.Toolkit.HighPerformance/Memory/Span2D{T}.cs

```diff
@@ -0,0 +1,1157 @@
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when one of the parameters are negative.</exception>
+        public unsafe Span2D(void* pointer, int height, int width, int pitch)
+        {
+            if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
+            {
+                ThrowHelper.ThrowArgumentExceptionForManagedType();
+            }
+
+            if (width < 0)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeExceptionForWidth();
+            }
+
+            if (height < 0)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeExceptionForHeight();
+            }
+
+            if (pitch < 0)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeExceptionForPitch();
+            }
+
+            OverflowHelper.EnsureIsInNativeIntRange(height, width, pitch);
+
+#if SPAN_RUNTIME_SUPPORT
+            this.span = new Span<T>(pointer, height);
+#else
+            this.Instance = null;
+            this.Offset = (IntPtr)pointer;
+            this.height = height;
+#endif
+            this.width = width;
+            this.Stride = width + pitch;
+        }
+
+#if !SPAN_RUNTIME_SUPPORT
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Span2D{T}"/> struct with the specified parameters.
+        /// </summary>
+        /// <param name="instance">The target <see cref="object"/> instance.</param>
+        /// <param name="offset">The initial offset within the target instance.</param>
+        /// <param name="height">The height of the 2D memory area to map.</param>
+        /// <param name="width">The width of the 2D memory area to map.</param>
+        /// <param name="pitch">The pitch of the 2D memory area to map.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal Span2D(object? instance, IntPtr offset, int height, int width, int pitch)
+        {
+            this.Instance = instance;
+            this.Offset = offset;
+            this.height = height;
+            this.width = width;
+            this.Stride = width + pitch;
+        }
+#endif
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Span2D{T}"/> struct.
+        /// </summary>
+        /// <param name="array">The target array to wrap.</param>
+        /// <param name="height">The height of the resulting 2D area.</param>
+        /// <param name="width">The width of each row in the resulting 2D area.</param>
+        /// <exception cref="ArrayTypeMismatchException">
+        /// Thrown when <paramref name="array"/> doesn't match <typeparamref name="T"/>.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// Thrown when either <paramref name="height"/> or <paramref name="width"/> are invalid.
+        /// </exception>
+        /// <remarks>The total area must match the lenght of <paramref name="array"/>.</remarks>
```

Comment:
Came across this typo in the documentation comments: `lenght`.
From a rudimentary search, it also appears in a few other locations.

Reply:
Whoops, thanks! Fixed in 7738d344df1002f6fbd206b4e54524e41a036c2a 👍